### DISCUSSION TITLE
Separate server for elsa designer vs execution

### DISCUSSION THREAD

**Discussion by sjd2021** [elsa-workflows/elsa-core#3494] [Q&A]
Due to certain limiting factors (time, security, etc.), I would like to have the elsa designer hosted in an existing web application. I have a separate server (set of distributed servers that are already configured for locking, etc.) where I'd like to offload all of the workflow execution responsibilities. One option I've considered is to recreate all of the endpoints needed to support the designer and forward those onto the execution server.

I guess I'll have control over which server initially gets the execute request, but I'm not sure how to make sure that the "designer-only" project isn't picking up any work from a timer job or anything like that.

**Comment by sjd2021** [comment]
Maybe there's a way to denote a particular server as a worker node that I'm missing?

**Comment by DanielMPries** [comment]
I think what you're looking for is actually [called out in the docs](https://elsa-workflows.github.io/elsa-core/docs/next/quickstarts/quickstarts-aspnetcore-server-dashboard#_hostcshtml--_viewimportscshtml).  Specifically you want to point your dashboard to a particular server instance in the `server-url` value of the `<esla-studio-root />`.

```cs
<elsa-studio-root server-url="https://your-elsa-server-url" monaco-lib-path="...">
    <elsa-studio-dashboard></elsa-studio-dashboard>
</elsa-studio-root>
```

**Comment by sjd2021** [comment]
@DanielMPries  Unfortunately, that's not the problem I'm facing. The site our users are accessing does not use bearer-auth, so I don't have a bearer token that I can forward to the workflow server. That site is also on a different domain, so cookie authentication has caused us some trouble. We have possible workarounds, but it'd be nice if for simplicity sake we could simply designate that one server is good for all but workflow execution and cleanup activities.

**Comment by sjd2021** [comment]
For now, I've set up an endpoint which solely exists to intercept requests from the designer to the actual WF server. That's working well enough for us.
